### PR TITLE
add htmlhint to lint html files

### PR DESCRIPTION
**Pull request by doshprompt**
[Review on Reviewable](https://reviewable.io/reviews/google/web-starter-kit/730)

### CODE REVIEW

**Comment by nicolasgarnier** [comment]
Thanks for this @doshprompt!
Could you please rebase? A PR we merged yesterday introduced a conflict.
Besides that this LGTM but I'm letting @addyosmani have a look before moving forward.

**Comment by doshprompt** [comment]
@addyosmani @nicolasgarnier sorry seems like I messed up the PR and submitted someone else's commits along with mine, should I push again or make a new PR?

**Comment by nicolasgarnier** [comment]
Feel free to start a new PR if you're having too many issues rebasing this one.

**Comment by doshprompt** [comment]
@nicolasgarnier fixed! Sorry it took so long :)

**Comment by doshprompt** [comment]
@addyosmani can you take a final look please as per @nicolasgarnier's suggestion? Thanks!

**Comment by addyosmani** [comment]
I haven't used htmlhint before and will need to double check on the value-add here and general maintainance/usefulness of the tool before we proceed. Really appreciate you putting together this PR @doshprompt :star:

**Comment by doshprompt** [comment]
@addyosmani thanks,! I will wait for you to look into htmlhint and get back to me on this. In general, it's the more useful bits of w3c as it warns about things like missing matching tags (unclosed divs etc.) or (<button>...</a>) which can be useful especially in large pages with a lot of markup.

File: .htmlhintrc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  "attr-value-double-quotes": true,
+  
+  // Attributes cannot be duplicated on a single tag
+  "attr-no-duplication": true,
+  
+  // Don't look for doctypes in the html files
+  "doctype-first": false,
+  
+  // Tags must be paired
+  "tag-pair": true,
```
Comment (Garbee) [inline .htmlhintrc]: I feel this should be false by default if included. There are many elements (`<li>` comes to mind right away) where the closing tag is not required. Forcing this should be opt-in for projects based on their style.

File: .htmlhintrc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  
+  // Empty tags must be closed by self
+  "tag-self-close": true 
+  
+  // All special characters must be escaped
+  "spec-char-escape": true,
```
Comment (Garbee) [inline .htmlhintrc]: Will this cause an issue with say `&` being  on its own. Say `Person & Thing`? If it is going to force escaping that, then this should be false by default. That usage is perfectly valid  and we shouldn't force over-escaping without a good reason.

File: .htmlhintrc [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  // Values for all SRC's in <img>|<script>|<link> tags must be set
+  "src-not-empty": true,
+  
+  /*
+   * PERFORMANCE
+   * =================
+   */
+  
+  // Disallow script tags in <head>
+  "head-script-disabled": true,
```
Comment (Garbee) [inline .htmlhintrc]: There are a *few* occasions where scripts in the head  are required (say polyfilling things for older IE.) This should be false by default and encouraged to be turned on if a site knows they don't need it.

File: .htmlhintrc (hunk 1 above)
Comment (doshprompt) [inline .htmlhintrc]: I guess that's fair, at least until we can fix the htmlhint rule upstream to heed these nuances of the official html spec.

File: .htmlhintrc (hunk 2 above)
Comment (doshprompt) [inline .htmlhintrc]: I don't believe it forces that, if you look at the rule, it's actually quite simplistic

https://github.com/yaniswang/HTMLHint/blob/master/src/rules/spec-char-escape.js#L12

In fact, I think it deserves another PR upstream to add more special characters that require to be escaped.

File: .htmlhintrc (hunk 3 above)
Comment (doshprompt) [inline .htmlhintrc]: While I agree with this in principal, I think the argument can be made that by default it should be turned on unless the site knows they need it.

I would prefer that the devs get warned about a head script, and then they go turn off this feature if they absolutely must have the script in HEAD, instead of carelessly putting a script in HEAD which is generally agreed to be a bad practice. Often times the script is not necessary in HEAD but someone just put it there by mistake etc.

As for IE polyfills, I think the official browser support of IE9+ means that there will be few and far between polyfills.

What do you think? Does this approach or my thinking/reasoning behind it at least make more sense?
